events/events.c

```diff
@@ -79,7 +83,8 @@ doevent(struct eventrec * r)
  * associated with expired timers registered via events_timer_register will
  * be run.  If any event function returns a non-zero result, no further
  * events will be run and said non-zero result will be returned; on error,
- * -1 will be returned.
+ * -1 will be returned.  May be interrupted by events_interrupt, in which
+ * case -2 will be returned.
```

Comment:
Is -2 a decent value to return?  Returning 0 doesn't fly with `events_spin()`, and returning `>0` seems to indicate an error in the event handler.

Reply:
`0` is the right value to return -- exiting the loop because we asked to exit the loop is not an error.  Just have `events_spin` check the flag.